assets/javascripts/discourse/initializers/chat-setup.js

```diff
@@ -124,6 +127,156 @@ export default {
           }
         });
       });
+      api.addSidebarSection(
+        (BaseCustomSidebarSection, BaseCustomSidebarSectionLink) => {
+          const SidebarChatSectionLink = class extends BaseCustomSidebarSectionLink {
+            constructor({ channel }) {
+              super(...arguments);
+              this.channel = channel;
+            }
+
+            get name() {
+              return this.channel.chatable_id;
+            }
+
+            get route() {
+              return "chat.channel";
+            }
+
+            get model() {
+              return {
+                chatChannel: this.channel,
+                channelId: this.channel.id,
+                channelTitle: this.channel.chatable.slug,
+              };
+            }
+
+            get title() {
+              return this.channel.title;
+            }
+
+            get text() {
+              return this.channel.title;
+            }
+
+            get prefixIcon() {
+              return "hashtag";
+            }
+
+            get prefixIconColor() {
+              return this.channel.chatable.color;
+            }
+
+            get prefixIconBadge() {
+              return this.channel.chatable.read_restricted && "lock";
+            }
+
+            get suffixIcon() {
+              return this.channel.unread_count > 0 && "circle";
+            }
+
+            get suffixCssClass() {
+              return this.channel.unread_mentions > 0 ? "urgent" : "unread";
+            }
+          };
+
+          const SidebarChatSection = class extends BaseCustomSidebarSection {
+            @tracked sectionLinks = A([]);
+
+            constructor({ sidebar }) {
+              super(...arguments);
+
+              this.sidebar = sidebar;
+              this.chatService = container.lookup("service:chat");
```

Comment:
We can't use @service here ? And if we can't shouldn't we load our own appEvents ?

Reply:
Service didn't work. Own appEvents is loaded 👍